examples/ct/handlers.go

```diff
@@ -283,33 +283,54 @@ func addChainInternal(ctx context.Context, c LogContext, w http.ResponseWriter,
 		return http.StatusBadRequest, fmt.Errorf("failed to verify add-chain contents: %v", err)
 	}
 
-	// Build up the SCT and MerkleTreeLeaf. The SCT will be returned to the client and
-	// the leaf will become part of the data sent to the backend.
+	// This is now.
+	timestamp := uint64(c.TimeSource.Now().UnixNano() / millisPerNano)
+
+	// Build the MerkleTreeLeaf that gets sent to the backend, and make a trillian.LogLeaf for it.
 	var issuer *x509.Certificate
 	if len(chain) > 1 {
 		issuer = chain[1]
 	}
-	merkleLeaf, sct, err := signerFn(c.logKeyManager, chain[0], issuer, c.TimeSource.Now())
+	merkleLeaf, err := makeLeafFn(chain[0], issuer, timestamp)
 	if err != nil {
-		return http.StatusInternalServerError, fmt.Errorf("failed to build SCT and Merkle leaf: %v %v", sct, err)
+		return http.StatusBadRequest, fmt.Errorf("failed to build MerkleTreeLeaf: %v", err)
 	}
-
-	// Send the Merkle tree leaf on to the Log server.
 	leaf, err := buildLogLeafForAddChain(c, *merkleLeaf, chain)
 	if err != nil {
 		return http.StatusInternalServerError, fmt.Errorf("failed to build LogLeaf: %v", err)
 	}
+
+	// Send the Merkle tree leaf on to the Log server.
 	req := trillian.QueueLeavesRequest{LogId: c.logID, Leaves: []*trillian.LogLeaf{&leaf}}
 
 	glog.V(2).Infof("%s: %s => grpc.QueueLeaves", c.LogPrefix, method)
-	_, err = c.rpcClient.QueueLeaves(ctx, &req)
+	rsp, err := c.rpcClient.QueueLeaves(ctx, &req)
 	glog.V(2).Infof("%s: %s <= grpc.QueueLeaves err=%v", c.LogPrefix, method, err)
 	if err != nil {
 		return http.StatusInternalServerError, fmt.Errorf("backend QueueLeaves request failed: %v", err)
 	}
+	if rsp == nil {
```

Comment:
Yes, I agree that both `rsp==nil` and `len(rsp) != len(req)` are impossible responses, but I think the defense is worthwhile.